tests/smoke_test.go

```diff
@@ -452,3 +452,40 @@ func TestPluginRun(t *testing.T) {
 	_, stderr = e.RunCommandExpectError("pulumi", "plugin", "run", "--kind=resource", "random", "--", "--help")
 	assert.Contains(t, stderr, "flag: help requested")
 }
+
+//nolint:paralleltest // disabled parallel because we change the plugins cache
```

Comment:
I don't think this is necessary, and we should be able to run this test in paralllel with others. We are running this using `ptesting.NewEnvironment`, which uses a temp directory, and sets `PULUMI_HOME` to that temp directory. Plugins should be installed into `PULUMI_HOME`, so it shouldn't interfere with the global state.  

(I do see we do something similar in `smoke_test.go`, but the change to set a new `PULUMI_HOME` for every test is relatively recent, so that was still necessary at the time. See https://github.com/pulumi/pulumi/pull/15559)